src/OpenLoco/src/Objects/Object.h

```diff
@@ -48,6 +48,23 @@ namespace OpenLoco
     constexpr size_t kMaxObjectTypes = 34;
     constexpr uint8_t kCargoTypeNull = 0xFF;
 
+    enum class SourceGame : uint8_t
+    {
+        // See below note for vanilla.
+        // If object SourceGame is custom then object header will be fully compared
+        // when looking for matches (i.e. takes into account object header checksum).
+        // This means that non-custom objects might have different versions but still
+        // be considered as the same object.
+        custom,
+        data, // tbc?
+
+        // Most custom objects set this, so can't be trusted to be only on vanilla.
+        // Use the isVanilla() function to actually check for custom as that does
+        // a lookup against the vanilla object list
+        vanilla,
+        openLoco,
```

Comment:
I'd feel better if these got some explicit number assigned to them, just in case.

Reply:
This is a good idea since these numbers won't ever change